app/src/main/java/com/beemdevelopment/aegis/ui/MainActivity.java

```diff
@@ -488,6 +495,78 @@ private void startAssignIconsActivity(List<VaultEntry> entries) {
         assignIconsResultLauncher.launch(assignIconIntent);
     }
 
+    private void startAssignGroupsDialog(List<VaultEntry> entries) {
+        View view = LayoutInflater.from(this).inflate(R.layout.dialog_select_group, null);
+        AutoCompleteTextView groupsSelection = view.findViewById(R.id.group_selection_dropdown);
+        TextInputLayout newGroupLayout = view.findViewById(R.id.text_group_name_layout);
+        TextInputEditText newGroupText = view.findViewById(R.id.text_group_name);
+
+        Collection<VaultGroup> groups = _vaultManager.getVault().getUsedGroups();
+        ArrayList<VaultGroupModel> groupsArray = new ArrayList<>();
+        groupsArray.add(new VaultGroupModel(getString(R.string.new_group)));
+        groupsArray.addAll(groups.stream().map(VaultGroupModel::new).collect(Collectors.toList()));
+
+        ArrayAdapter<VaultGroupModel> adapter = new ArrayAdapter(this, android.R.layout.simple_dropdown_item_1line, groupsArray) {
+            @NonNull
+            @Override
+            public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+                View view = super.getView(position, convertView, parent);
+                TextView textView = view.findViewById(android.R.id.text1);
+                textView.setText(getItem(position).toString());
+                return view;
+            }
+        };
+
+        groupsSelection.setAdapter(adapter);
+        groupsSelection.setOnItemClickListener((parent, view1, position, id) -> {
+            _selectedVaultGroupModel = (VaultGroupModel) parent.getItemAtPosition(position);
+            if(_selectedVaultGroupModel.isPlaceholder()) {
+                newGroupLayout.setVisibility(View.VISIBLE);
+                newGroupText.requestFocus();
+            } else {
+                newGroupLayout.setVisibility(View.GONE);
+            }
+        });
+
+        AlertDialog dialog = new MaterialAlertDialogBuilder(this)
+                .setTitle(R.string.assign_groups)
+                .setView(view)
+                .setPositiveButton(android.R.string.ok, null)
+                .setNegativeButton(android.R.string.cancel, null)
+                .create();
+
+        dialog.setOnShowListener(d -> {
```

Comment:
Do we need this trick to find the positive button or can we pass the listener to ``setPositiveButton`` in this case?